pump_test.go

```diff
@@ -1,29 +1,83 @@
 package main
 
 import (
+	"os"
+	"path/filepath"
+	"strconv"
 	"sync"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	"github.com/INFURA/ipfs-pump/pump"
 )
 
 func TestPump(t *testing.T) {
-	testPump(t, 50, 1)
-	testPump(t, 50, 50)
-	testPump(t, 50, 100)
-	testPump(t, 500, 10)
+	testPump(t, pump.NewMockDrain(), 50, 0, 1)
+	testPump(t, pump.NewMockDrain(), 50, 0, 50)
+	testPump(t, pump.NewMockDrain(), 50, 0, 100)
+	testPump(t, pump.NewMockDrain(), 500, 0, 10)
+
+	testPump(t, pump.NewMockFailingDrain(5), 10, 5, 1)
+	testPump(t, pump.NewMockFailingDrain(50), 100, 50, 50)
+	testPump(t, pump.NewMockFailingDrain(50), 50, 50, 100)
+	testPump(t, pump.NewMockFailingDrain(500), 5000, 500, 100)
 }
 
-func testPump(t *testing.T, count int, worker uint) {
+func testPump(t *testing.T, drain pump.Drain, count int, failedCount uint, worker uint) {
+	tmpFailedBlocksFile := filepath.Join(os.TempDir(), strconv.Itoa(int(time.Now().Unix()+int64(count)+int64(failedCount)+int64(worker))))
```

Comment:
https://golang.org/pkg/io/ioutil/#TempFile maybe?

Reply:
That would be a good idea for sure but I want to avoid actually creating the file. Just construct a filepath where the file should be created. Do we have a method like to disposition somewhere?

Reply:
ha yeah, make sense